src/OneExplorer/ConfigObject.ts

```diff
@@ -122,21 +126,33 @@ export class ConfigObj {
     return found ? true : false;
   }
 
+  /**
+   * @brief getter for config setting
+   */
+  public get getConfigSetting(): ConfigSetting {
+    let configSetting: ConfigSetting = new OneConfigSetting();
```

Comment:
this code is must be update to switch-case because ConfigSetting is created twice

Reply:
Please refer to the "NodeFactory" code in "OneExplorer".